File: 8kyu/beginner-series-2-clock/beginner-series-2-clock.js
function past(h, m, s) {
	const times = [h, m, s];
	const total = times.reduce((acc, time) => (acc = 60 * acc + time), 0);
	return total * 1000;
}

// describe(" convert time to milliseconds", () => {
// 	const cases = [
// 		[0, 1, 1, 61000],
// 		[1, 1, 1, 3661000],
// 		[0, 0, 0, 0],
// 		[1, 0, 1, 3601000],
// 		[1, 0, 0, 3600000],
// 	];
// 	for (const [h, m, s, output] of cases) {
// 		it(`should return ${output} when h = ${h}, m = ${m}, and s = ${s}`, () => {
// 			expect(past(h, m, s)).toBe(output);
// 		});
// 	}
// });

File: 8kyu/beginner-series-2-clock/beginner-series-2-clock.go
package main

import (
	"fmt"
)

// func Past(h, m, s int) int {
// 	times := []int{h * 60 * 60, m * 60, s}
// 	total := 0
// 	for _, time := range times {
// 		total += time
// 	}
// 	return total * 1000
// }

func Past(h, m, s int) int {
	times := []int{h * 60 * 60, m * 60, s}
	total := 0
	for i := 0; i < len(times); i += 1 {
		total += times[i]
	}

	return total * 1000
}

func main () {
	fmt.Println(Past(0, 1, 1))
	fmt.Println(Past(1, 1, 1)) 
}